config/confighttp/README.md

```diff
@@ -20,6 +20,9 @@ README](../configtls/README.md).
 - [`read_buffer_size`](https://golang.org/pkg/net/http/#Transport)
 - [`timeout`](https://golang.org/pkg/net/http/#Client)
 - [`write_buffer_size`](https://golang.org/pkg/net/http/#Transport)
+- `compression`: Compression type to use among `zstd`, `snappy`, `gzip`, `zlib`, and `deflate`.
+  - `zstd` and `snappy` are recommended.
+  - `none` will be treated as uncompressed, and any other inputs will cause an error.
```

Comment:
Why zstd and snappy  are recommended? Do we really need to make a recommendation? Why not the most common one gzip?

Reply:
This change was based on @jpkrohling's [comment](https://github.com/open-telemetry/opentelemetry-collector/pull/4441#pullrequestreview-814747382):

> Should we be making a recommendation here to users? According to the notes from the SIG meeting from Nov 3rd, zstd would be the 1st option, with snappy being the 2nd.

Reply:
As a user of this, I would appreciate being given a hint on what's the current best/preferred algorithm. I don't know why zstd and snappy are recommended, those were the ones mentioned as being "the first choice and second choice" during the SIG meeting where we discussed this, according to the meeting minutes.

Reply:
@jpkrohling that is true they were the first and second to add on top of "gzip". So I think gzip is actually the top.

Reply:
As per this issue #2548,
> Gzip is CPU intensive and offers less of a compression benefit than more modern compression like snappy or zstd

I think the reason we introduced `snappy` and `zstd` is due to this downside of `gzip`.

Reply:
I will change the recommendation part to 
> look at the documentation for the server-side of the communication

As we discussed in the SIG meeting on Dec 8